modules/cms-kit/src/Volo.CmsKit.MongoDB/Volo/CmsKit/MongoDB/Blogs/MongoBlogPostRepository.cs

```diff
@@ -48,21 +52,33 @@ public async Task<int> GetCountAsync(Guid blogId, CancellationToken cancellation
         {
             var token = GetCancellationToken(cancellationToken);
             var dbContext = await GetDbContextAsync(token);
-            var blogPostQueryable = await WithDetailsAsync();
+            var blogPostQueryable = dbContext.Collection<BlogPost>().AsQueryable();
+            var usersQueryable = dbContext.Collection<CmsUser>().AsQueryable();
 
-            var queryable = blogPostQueryable
+            IQueryable<BlogPost> queryable = blogPostQueryable
```

Comment:
`var`

Reply:
I'm avoiding to cast `IQueryable<BlogPost>`. Because return type is `IMongoQueryable<BlogPost>`. If it's better, I can use `var` keyword and add a casting. @ilkayilknur ?

Reply:
Then you should define it as `IMongoQueryable<BlogPost>`.

Reply:
It's required to be `IQueryable<T>` to call `OrderBy` method with string parameter from `System.Linq.Dynamic.Core` @ilkayilknur